TableBookingAPI/TableBooking/Services/BookingService.cs

```diff
@@ -86,9 +86,25 @@ public async Task<IActionResult> GetAllBookings(Guid userId)
             return new OkObjectResult(bookings);
         }
 
-        public async Task<IActionResult> UpdateBookingAsync(UpdateBookingDto updateBookingDto, Guid userId)
+        public async Task<IActionResult> UpdateBookingAsync(UpdateBookingDto updateBookingDto, Guid userId, Guid bookingId)
         {
-            return new OkObjectResult("XD");
+            var booking = await _unitOfWork.BookingRepository.GetBookingByIdForSpecificUserAsync(bookingId, userId);
+            if (booking == null)
+                return new BadRequestObjectResult($"Booking with id {bookingId} doesn't exist.");
+            
+            var newBooking = new Booking
+            {
+                Id = booking.Id,
+                Date = updateBookingDto.Date,
+                DurationInMinutes = updateBookingDto.DurationInMinutes,
+                AmountOfPeople = updateBookingDto.AmountOfPeople,
+                TableId = booking.TableId,
+                AppUserId = userId
+            };
+
+            await _unitOfWork.BookingRepository.Update(newBooking);
+            
```

Comment:
there should be await _unitOfWork.SaveChangesAsync(); instead of directly from repository